python/lsst/analysis/tools/tasks/sourceObjectTableAnalysis.py

```diff
@@ -0,0 +1,115 @@
+from __future__ import annotations
+
+__all__ = ("SourceObjectTableAnalysisConfig", "SourceObjectTableAnalysisTask")
+
+import numpy as np
+import pandas as pd
+from astropy.table import vstack
+from lsst.pipe.base import connectionTypes as ct
+from smatch import Matcher
+
+from ..interfaces import AnalysisBaseConfig, AnalysisBaseConnections, AnalysisPipelineTask
+
+
+class SourceObjectTableAnalysisConnections(
+    AnalysisBaseConnections,
+    dimensions=("visit",),
+    defaultTemplates={
+        "inputName": "sourceTable_visit",
+        "inputCoaddName": "deep",
+        "associatedSourcesInputName": "isolated_star_sources",
+        "outputName": "sourceObjectTable",
+    },
+):
+    data = ct.Input(
+        doc="Visit based source table to load from the butler",
+        name="sourceTable_visit",
+        storageClass="ArrowAstropy",
+        dimensions=("visit", "band"),
+        deferLoad=True,
+    )
+
+    associatedSources = ct.Input(
+        doc="Table of associated sources",
+        name="{associatedSourcesInputName}",
+        storageClass="ArrowAstropy",
+        multiple=True,
+        deferLoad=True,
+        dimensions=("instrument", "skymap", "tract"),
+    )
+
+    refCat = ct.Input(
+        doc="Catalog of positions to .",
+        name="objectTable",
+        storageClass="DataFrame",
+        dimensions=["skymap", "tract", "patch"],
+        multiple=True,
+        deferLoad=True,
+    )
+
+
+class SourceObjectTableAnalysisConfig(
+    AnalysisBaseConfig, pipelineConnections=SourceObjectTableAnalysisConnections
+):
+    pass
+
+
+class SourceObjectTableAnalysisTask(AnalysisPipelineTask):
+    ConfigClass = SourceObjectTableAnalysisConfig
+    _DefaultName = "sourceTableVisitAnalysis"
+
+    def runQuantum(self, butlerQC, inputRefs, outputRefs):
+        inputs = butlerQC.get(inputRefs)
+
```

Comment:
To get a more complete plotInfo box on the plot, suggest
```
dataId = butlerQC.quantum.dataId
plotInfo = self.parsePlotInfo(inputs, dataId)
```
is added here and the run command on line 114 is changed to:
`outputs = self.run(data=allCat, bands=band, plotInfo=plotInfo)`
.